op-node/node/node.go

```diff
@@ -293,9 +293,13 @@ func (n *OpNode) initRuntimeConfig(ctx context.Context, cfg *Config) error {
 
 func (n *OpNode) initL1BeaconAPI(ctx context.Context, cfg *Config) error {
 	if cfg.Beacon == nil {
-		n.log.Warn("No beacon endpoint configured. Configuration is mandatory for the Ecotone upgrade")
+		if cfg.Rollup.EcotoneTime != nil {
+			n.log.Warn("No beacon endpoint configured. Configuration is mandatory for the Ecotone upgrade", "ecotoneTime", *cfg.Rollup.EcotoneTime)
+		}
```

Comment:
Since `L1BeaconEndpointConfig.Check()` already enforces a value to be set, any value, I think we should return an error here if the beacon setup is completely missing.

Reply:
Should we add the same `l1.beacon.ignore` check here. Error it by default, with a way for node operators to override.

It would let people continue to run nodes without a beacon endpoint until Ecotone, but they would have to opt in and say I ignore this check so they would be at least have notice prior to Ecotone.

Reply:
Added a commit to try and cover the UX here: enforce beacon API endpoint configuration, retry on failures, handle missing endpoint, inform user of purpose, and allow ignoring while still having a non-nil client when continuing to run the op-node.